src/Microsoft.DotNet.Interactive.Formatting/HtmlFormatter{T}.cs

```diff
@@ -210,15 +226,45 @@ internal static HtmlFormatter<T> CreateTreeViewFormatterForAnyObject()
     {
         if (typeof(T).IsScalar())
         {
-            return new HtmlFormatter<T>((value, context) => { HtmlFormatter.FormatAndStyleAsPlainText(value, context); });
+            return new HtmlFormatter<T>((value, context) => HtmlFormatter.FormatAndStyleAsPlainText(value, context));
         }
 
-        var members = typeof(T).GetMembersToFormat()
-                               .GetMemberAccessors<T>();
+        (HtmlTag propertyLabelTdTag, Func<T, HtmlTag> getPropertyValueTdTag)[] rows =
+            typeof(T).GetMembersToFormat()
+                     .GetMemberAccessors<T>()
+                     .Select(a => (
+                                      new HtmlTag("td", a.MemberName),
+                                      new Func<T, HtmlTag>(obj => new HtmlTag("td", c =>
+                                      {
+                                          var value = a.GetValueOrException(obj);
+                                          value.FormatTo(c, HtmlFormatter.MimeType);
+                                      }))))
+                     .ToArray();
+
+        // represent IEnumerable as a separate special property "(values)" at the end of the list
+        if (typeof(T).IsEnumerable())
+        {
+            if (typeof(T).ShouldIncludePropertiesInOutput())
+            {
+                var enumerableFormatter = HtmlFormatter.GetDefaultFormatterForAnyEnumerable(typeof(T));
+
+                (HtmlTag propertyLabelTdTag, Func<T, HtmlTag> getPropertyValueTdTag) enumerableAccessor =
+                    (new HtmlTag("td", new HtmlTag("i", "(values)")), obj => new HtmlTag("td", c =>
+                        {
+                            enumerableFormatter.Format(obj, c);
+                        }));
+                Array.Resize(ref rows, rows.Length + 1);
+                rows[^1] = enumerableAccessor;
```

Comment:
minor: Would it be better to leave rows as an enumerable and use `Enumerable.Append` here (and call `ToArray` right before calling `BuildTreeView` below)?

Reply:
I would assume that would be less performant than doing the array operations directly, though it would be far from the most critical piece of code to optimize here.